### DISCUSSION TITLE
Welcome to WRC-TUG Discussions!

### DISCUSSION THREAD

**Discussion by darynwhite** [NOAA-PMEL/WRC-TUG#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by derekcoffman** [comment]
Hey Daryn, thanks for starting this discussion. I think github is a great platform for this and I'm tying to use if for more of my own project management type stuff. And the fact that I'm currently watching github universe probably doesn't hurt.

Anyway, a quick introduction. I'm Derek Coffman and I work at PMEL in the Atmospheric Chemistry group. Most of our research is looking at marine aerosols from ships but we also look at some land based station data as well. Also, we are developing instrument payloads for UAVs. We are primarily interested flying them from ships to augment our surface measurements but can also do land based work. 

As for TUG specific issues, I'm currently working on

- python based DAQ 
  - distributed DAQ using client/server system 
  - asyncio
  - websockets 
  - bokeh for data plotting
  - web front end for OS independence
- Raspberry PI and Arduino based instrument DAQ (used in UAV payloads)
- Data management - improve DAQ, analysis and archive (ERDDAP) workflow
- Data analysis - currently working in Igor Pro (similar to Matlab) but transitioning python based. Plotly/Dash, Holoviz

No specific questions at the moment looking forward to keeping these conversations going.

**Reply by darynwhite** [reply]
Thanks for kicking off introductions, Derek! I'm curious about your work on distributed DAQ with python. How far has that work come along?

**Comment by shaunwbell** [comment]
I use Github and didn't even know about "discussions" interesting.

I'm Shaun Bell - PMEL EcoFOCI group.  We spend a lot of time/energy on the physical oceanography of Alaskan waters (Gulf, Bering, Arctic) with collaborations into the biology and productivity via AFSC.  This means I manage the data flow from a wealth of sensors and vendors and my Processing/QC-QA/Visualization pathways are very Python heavy and include:
- visualization (matplotlib, altair, plotly)
- processing (pandas, xarray)
- analysis (TEOS seawater package, signal processing of various sorts)
- TODO: ML techniques for automated QC/QA (now is just a visual inspection in many ways)

I also manage a SQL database for our metadata, an internal ERDDAP server as part of our data workflow, a website for frontend and dynamic visualization (see altair/plotly) and am otherwise a bit of a computational swiss army knife in our group.

At home: but still TuG related
- many wifi enable ESP8266 microcontrollers and sparkfun arduino based sensors around my property
- RPi based data hosting and display (via html/php/javascript) - you know... for fun... whats more fun than having your own personal erddap server? ;p

I would love to be a better user of Github (mostly in the collaborative sense) and am about to go through the same Data Science Cert program that I Roman went through - so hopefully that will be a valuable resource.  And I think some grass roots software development at PMEL between group programmers (more along the lines of subroutines, modules, and guided workflows for instrument processing) could be incredibly beneficial.

**Reply by darynwhite** [reply]
Shaun, I couldn't agree more about the grassroots collaboration at PMEL! Heck, even across the WRC ;)

**Reply by darynwhite** [reply]
> I use Github and didn't even know about "discussions" interesting.

This is a new feature, not certain when it came into production.

**Reply by derekcoffman** [reply]
It was announced at GitHub universe on Tuesday so you were very bleeding
edge. Heh

**Comment by MCasari-PMEL** [comment]
Hi everyone,
![](https://avatars1.githubusercontent.com/u/16406932?s=400&u=a4851c68fc4e337aeeaeca54f9760974ccf22df5&v=4|width=100)

I'm Matt Casari from the PMEL Engineering Development Division.  Our work is mostly in support of the other groups needs to monitor climate, atmosphere and ocean, but we develop new systems and sensors as well.  I am an electronics engineer, but spend most of my time working with embedded software/firmware.  I have sort of a three-way split of my time based on what portion of the project we're working on that can be loosely broken down as follows:

### Electronics Design & Test
* System Design, Schematic Capture, Electronics Component Selection, Printed Circuit Board Layout
* Embedded Hardware Test (Oscilloscopes, Logic Analyzers, Signal Analyzers, Network Analyzers)

### Embedded Software Design & Test
* Embedded Software (C/C++, Arduino, Micropython)
* Embedded Software Testing (Unity, CMock)
* Documentation (Doxygen)

### System Test & Interface
* Data acquisition (pyserial, labjack, pyvisa)
* System Testing & automation (pytest, threading)
* Unit testing (also pytest)
* Processing (pandas)

In addition, I maintain EDD's GitHub repositories, create user manuals and other documentation (UML, markdown) and have automated a number of tasks in EDD using Google App Script.  I helped cofound the TuG with Shaun and Daryn because I enjoy talking about all sorts of tech topics and love learning new things.  I'm excited to see some new talks in 2021, and am also excited to talk about testing and pytest in February; a topic that I'd like to see get a lot more traction at PMEL.

**Comment by darynwhite** [comment]
Hey TuG!

I'm Daryn White from the Global Tropical Moored Buoy Array (I know, GTMBA is good) at PMEL. We're a long time project that started in the 1990s in the Tropical Pacific and has expanded into the Atlantic and Indian basins as well; for a complete understanding, look at [GTMBA's Mission](https://www.pmel.noaa.gov/gtmba/mission)

I work primarily as the QC reviewer for most of the data collected from our deployments, so similar to what @shaunwbell does for EcoFOCI, which is how I connected with him about this group. In addition, I support the group with:

- Database interfaces for instrument tracking/history
- Workflow development utilizing Python
- Calibration of our Atlas TC modules (interfacing w/Sea-Bird)
- Assisting with real-time data review & system statuses

There's probably more that comes across my plate any given year, but that's enough for now. 

I'm most excited about TUGs ability to introduce individuals with similar roles and/or workflows across the whole of NOAA's WRC and to topics that otherwise feel unreachable to some.

I have a one-year-old toddling around now, so time for hobbies is *hopefully* in my future but lacking currently.
